### PR TITLE
feat(core) add macOS arm support

### PR DESCRIPTION
**Pull request by eyeofstorm**
## Changes Proposed:
!!! This PR based on PR #14423. !!! 
Thanks @motivewc for the original code.

-  Use ARM friendly replacements for AVX instructions, fixes argon2 compile.

## Tests Performed:

- Built from source on macOS Ventura 13.2 on M1 Mac.
- Built from source on Windows 11.

## How to Test the Changes:

Build and run.

## Known Issues and TODO List:

No known issues.
Tested on M1 Mac and Windows 11.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
Adding run build label to see how this turns out.

**Comment by acidmanifesto** [comment]
i am going to summon shin for this one as i am not a expert with anything with mac. but current workflow shows window passing which was not the case with the past closed pr.

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
> i am going to summon shin for this one as i am not a expert with anything with mac. but current workflow shows window passing which was not the case with the past closed pr.

That is because this PR solves the issue by checking if _M_X64 is defined, which is part of MSVC

File: deps/argon2/argon2/blake2/blamka-round-opt.h [hunk 1]
```diff
@@ -20,12 +20,17 @@
...
+#elif defined(__aarch64__) || defined(_M_ARM64)
+#include "sse2neon.h"
+#endif
+
 #if defined(__SSSE3__)
 #include <tmmintrin.h> /* for _mm_shuffle_epi8 and _mm_alignr_epi8 */
 #endif
 
-#if defined(__XOP__) && (defined(__GNUC__) || defined(__clang__))
+#if (defined(__x86_64__) || defined(_M_X64) || defined(_M_AMD64)) && defined(__XOP__) && (defined(__GNUC__) || defined(__clang__))
```
Comment (Kitzunu) [inline deps/argon2/argon2/blake2/blamka-round-opt.h:33]: Actually, why do you check the windows variable here?

**Comment by Kitzunu** [comment]
CI passed and author tested it.
That is fine for me, as it is only adding a file in deps and doesnt change anything existing at all 👍

File: deps/argon2/argon2/blake2/blamka-round-opt.h (hunk 1 above)
Comment (eyeofstorm) [reply deps/argon2/argon2/blake2/blamka-round-opt.h:33]: not only the windows. it check wether in x86_64 or not. linux will also hit this condition.

Comment (Kitzunu) [reply deps/argon2/argon2/blake2/blamka-round-opt.h:33]: yes but _M_X64 and _M_AMD64 are for MSVC and not linux, no? So why would you need to check for them here when we dont previously do that

Comment (eyeofstorm) [reply deps/argon2/argon2/blake2/blamka-round-opt.h:33]: Yes, you are right. x86intrin.h is for gcc/clang. MSVC uses intrin.h.
I will fix this. only __x86_64__ need to be check.

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
Looks good to me

**Pull request merged**